npe2/_plugin_manager.py

```diff
@@ -313,7 +309,7 @@ def get_writer(
 
         for writer in self.iter_compatible_writers(layer_types):
             if plugin_name and not writer.command.startswith(plugin_name):
-                continue
+                continue  # pragma: no cover
```

Comment:
This will get hit when a user has multiple single-image-layer writers installed (from different plugins) and is querying by plugin_name.

Right now I believe the testing is focused on environments with just one test plugin installed. Should we add another to cover this?

Reply:
the logic of simply continuing is simple enough.  But I refactored out the `continue` and the pragma with `if not plugin_name or writer.command.startswith(plugin_name):`